src/traces/scattercarpet/plot.js

```diff
@@ -37,6 +38,6 @@ module.exports = function plot(gd, plotinfoproxy, data) {
         // separately to all scattercarpet traces, but that would require
         // lots of reorganization of scatter traces that is otherwise not
         // necessary. That makes this a potential optimization.
-        node.attr('clip-path', 'url(#clip' + carpet.uid + 'carpet)');
+        Drawing.setClipUrl(node, carpet._clipPathId);
```

Comment:
yeah, as mentioned in https://github.com/plotly/plotly.js/pull/1815#discussion_r124943024

Tried to figure out if we have a way to enforce that... only allow `attr('clip-path', ...)` inside `Drawing.setClipUrl` - like something to add to `strict-d3`? Seemed pretty invasive and hacky though.